frcxx_nft/src/state.rs

```diff
@@ -0,0 +1,207 @@
+//! Abstraction of the on-chain state related to NFT accounting
+use cid::multihash::Code;
+use cid::Cid;
+use fvm_ipld_amt::Amt;
+use fvm_ipld_amt::Error as AmtError;
+use fvm_ipld_blockstore::Block;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::tuple::*;
+use fvm_ipld_encoding::CborStore;
+use fvm_ipld_encoding::DAG_CBOR;
+use fvm_ipld_hamt::Error as HamtError;
+use fvm_ipld_hamt::Hamt;
+use fvm_shared::ActorID;
+use thiserror::Error;
+
+pub type TokenID = u64;
+
+#[derive(Error, Debug)]
+pub enum StateError {
+    #[error("ipld amt error: {0}")]
+    IpldAmt(#[from] AmtError),
+    #[error("ipld hamt error: {0}")]
+    IpldHamt(#[from] HamtError),
+    #[error("other error: {0}")]
+    Other(String),
+}
+
+/// Each token stores its owner, approved operators etc.
+#[derive(Serialize_tuple, Deserialize_tuple, Clone, Debug)]
+pub struct TokenData {
+    pub owner: ActorID,
+    // operators on this token
+    pub approved: Vec<ActorID>, // or maybe as a Cid to an Amt
+    pub metadata_uri: String,
```

Comment:
Assuming the spec will not require URI, use `metadata_key` or `metadata_id` or similar.